lib/src/event.dart

```diff
@@ -349,13 +349,38 @@ class Event extends MatrixEvent {
   /// Removes an unsent or yet-to-send event from the database and timeline.
   /// These are events marked with the status `SENDING` or `ERROR`.
   /// Throws an exception if used for an already sent event!
-  Future<void> cancelSend() async {
+  ///
+  Future<void> cancelSend({
+    MatrixLocalizations i18n = const MatrixDefaultLocalizations(),
+  }) async {
     if (status.isSent) {
       throw Exception('Can only delete events which are not sent yet!');
     }
 
     await room.client.database?.removeEvent(eventId, room.id);
+
+    if (room.lastEvent?.eventId == eventId) {
+      room.lastEvent = Event(
```

Comment:
that's not how I suggested it actually. I suggested to **redact** the lastEvent with a fakeEvent, not replace it. Then when it comes to the logic to generate the localizedBody of a redacted event, check the sending status and if it is not synced, then return `l10n. cancelledSend`

Reply:
I don't see how that would affect the outcome 🤔 . `setRedactedEvent` just sets the redacted_because unsigned property and then we return a custom `Redacted` text from the .body getter. That is much more code to navigate rather than just having the .cancelSend() method set the 'Cancelled sending message' body.